### PR TITLE
[Link] Inline link not accessible on Android

### PR DESCRIPTION
**Pull request by ayush547**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [x] android

### Description of changes

Inline links <Text> inside <Text> are not keyboard accessible, this is a bug from RN Core, issues - [32004](https://github.com/facebook/react-native/issues/32004), [35194](https://github.com/facebook/react-native/issues/35194). 
Updating inline sample to showcase a work around this.

### Verification

Verified keyboard focus and talkback support.

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [x] Keyboard Accessibility
- [x] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/components/Link/src/Link.tsx [hunk 1]
```diff
@@ -39,8 +39,9 @@ export const Link = compose<LinkType>({
     return (final: LinkProps, ...children: React.ReactNode[]) => {
       // the event fires twice due to native's implementation of inline link
       const { inline, ...mergedProps } = mergeProps(link.props, final);
+      const supportsInlineLink = Platform.OS !== 'android';
```
Comment (ankraj12) [inline packages/components/Link/src/Link.tsx:43]: Android supports inline links, this name is misleading in that case

**Review by ankraj12 — APPROVED** [review]

File: packages/components/Link/src/Link.tsx (hunk 1 above)
Comment (ayush547) [reply packages/components/Link/src/Link.tsx:43]: Updated

File: apps/fluent-tester/src/TestComponents/LinkV1/InlineLinksTest.android.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { View, Alert, StyleSheet } from 'react-native';
+
+import { LinkV1 as Link, TextV1 as Text } from '@fluentui/react-native';
+import { Stack } from '@fluentui-react-native/stack';
+
+import { stackStyle } from '../Common/styles';
+
+export const InlineLinks: React.FunctionComponent = () => {
+  const doPress = React.useCallback(() => Alert.alert('Alert.', 'You have been alerted.'), []);
+  const doAllyTap = React.useCallback(() => Alert.alert('Alert.', 'You have invoked onAllyTap.'), []);
```
Comment (rohanprasadofficial) [inline apps/fluent-tester/src/TestComponents/LinkV1/InlineLinksTest.android.tsx:11]: nit :   Ally -> A11y

**Review by rohanprasadofficial — APPROVED** [review]
Added a small suggestion

File: packages/components/Link/src/Link.tsx [hunk 3]
```diff
@@ -39,8 +39,9 @@ export const Link = compose<LinkType>({
     return (final: LinkProps, ...children: React.ReactNode[]) => {
       // the event fires twice due to native's implementation of inline link
       const { inline, ...mergedProps } = mergeProps(link.props, final);
+      const supportsTextInText = Platform.OS !== 'android';
 
-      return inline || mergedProps.selectable ? (
+      return supportsTextInText && (inline || mergedProps.selectable) ? (
```
Comment (rurikoaraki) [inline packages/components/Link/src/Link.tsx:44]: This means that inline links are not supported on Android. Is that the intended change? If so you should update the docs and the comment of the inline prop to reflect this.

**Pull request merged**